File: pepole-microservice/src/main/java/com/example/demo/MongoConfig.java
package com.example.demo;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.env.Environment;
import org.springframework.data.mongodb.MongoDatabaseFactory;
import org.springframework.data.mongodb.core.MongoTemplate;
import org.springframework.data.mongodb.core.SimpleMongoClientDatabaseFactory;

import java.util.Objects;

@Configuration
public class MongoConfig {
    private final Environment env;

    public MongoConfig(Environment env) {
        this.env = env;
    }

    @Bean
    public MongoTemplate mongoTemplate() {
        return new MongoTemplate(mongoDatabaseFactory());
    }

    @Bean
    public MongoDatabaseFactory mongoDatabaseFactory() {
        return new SimpleMongoClientDatabaseFactory(Objects.requireNonNull(env.getProperty("spring.data.mongodb.uri")));
    }
}

File: vaultitem-service/src/main/kotlin/com/example/kotlinservice/MongoConfig.kt
package com.example.kotlinservice


import com.mongodb.client.MongoClient
import com.mongodb.client.MongoClients
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.mongodb.MongoDatabaseFactory
import org.springframework.data.mongodb.core.SimpleMongoClientDatabaseFactory
import org.springframework.data.mongodb.core.convert.MongoCustomConversions



@Configuration
class MongoConfig {

    @Value("\${spring.data.mongodb.uri}")
    lateinit var mongoDbUri: String

    @Bean
    fun mongoClient(): MongoClient {
        return MongoClients.create(mongoDbUri)
    }

    @Bean
    fun mongoCustomConversions() : MongoCustomConversions {
        return MongoCustomConversions(listOf(VaultItemReadConverter()))
    }

    @Bean
    fun mongoDbFactory(): MongoDatabaseFactory {
        return SimpleMongoClientDatabaseFactory(mongoClient(), getDatabaseName())
    }

    @Bean
    fun getDatabaseName(): String {
        return "app-db"
    }
}